cst438_gradebook/src/test/java/com/cst438/AssignmentControllerTests.java

```diff
@@ -0,0 +1,184 @@
+        response = mvc.perform(MockMvcRequestBuilders
+                        .get("/assignment/{assignment_id}", assignmentId)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
+    }
+    @Test
+    public void AddAssignmentTest() throws Exception {
+        AssignmentDTO assignmentDTO = new AssignmentDTO(1, "logic", "2023-09-28", "BUS 203 - Financial Accounting", 30157);
+
+        Assignment new_assignment = new Assignment();
+        new_assignment.setName(assignmentDTO.assignmentName());
+        new_assignment.setDueDate(Date.valueOf(assignmentDTO.dueDate()));
+        when(assignmentRepository.save(any(Assignment.class))).thenReturn(new_assignment);
+
+        MockHttpServletResponse response;
+        response = mvc.perform(MockMvcRequestBuilders.post("/assignment")
+                        .content(asJsonString(assignmentDTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse();
+
+        assertEquals(200, response.getStatus());
+    }
+
+
+
+
+
+    @Test
+    public void ListAssignmentsTest() throws Exception {
+
+        String instructorEmail = "dwisneski@csumb.edu";
+        Assignment assign1 = new Assignment();
+        assign1.setId(1);
+        assign1.setName("Rest assignment");
+        assign1.setDueDate(Date.valueOf("2023-09-23"));
+
+        Course course1 = new Course();
+        course1.setTitle("Software Engineering ");
+        course1.setCourse_id(1);
+        assign1.setCourse(course1);
+
+        Assignment assign2 = new Assignment();
+        assign2.setId(2);
+        assign2.setName("Modus tollens");
+        assign2.setDueDate(Date.valueOf("2023-10-27"));
+
+        Course course2 = new Course();
+        course2.setTitle("Reasoning with Logic");
+        course2.setCourse_id(2);
+        assign2.setCourse(course2);
+        List<Assignment> assignments = Arrays.asList(assign1, assign2);
+
+        when(assignmentRepository.findByEmail(instructorEmail)).thenReturn(assignments);
+        MockHttpServletResponse response = mvc.perform(MockMvcRequestBuilders.get("/assignment")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn().getResponse();
+
+        AssignmentDTO[] res = new AssignmentDTO[assignments.size()];
+        for (int i = 0; i < assignments.size(); i++) {
+            Assignment assignment = assignments.get(i);
+            res[i] = new AssignmentDTO(
+                    assignment.getId(),
+                    assignment.getName(),
+                    assignment.getDueDate().toString(),
+                    assignment.getCourse().getTitle(),
+                    assignment.getCourse().getCourse_id()
+            );
+        }
+        assertEquals(2, res.length);
```

Comment:
This test is good as you created multiple objects and then confirmed they were added using the assert statements. I see you followed this testing convention, because returning all objects may yield more problems then testing/adding for one. Even though it might seem redundant, the same format should be reflected in all tests where it is possible.